tests/test_parser.py

```diff
@@ -380,3 +454,46 @@ def test_tls_help(self):
         self.parser._set_defaults(None)
         help_output = self.parser._parser_tls.format_help()
         self.assert_snapshot('tls_help', help_output)
+
+
+class ParserModuleFunctionTestCase(unittest.TestCase):
+    # pylint: disable=protected-access
+    def test_create_parser(self):
+        description = 'parser description'
+        logfilename = 'logfilename'
+
+        parser = create_parser(description, logfilename)
+
+        self.assertTrue(isinstance(parser, CliParser))
+        self.assertEqual(parser._logfilename, logfilename)
+        self.assertEqual(parser._bootstrap_parser.description, description)
+
+    @mock.patch('gvmtools.parser.TLSConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_unix_socket_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test()
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_tls_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test('tls', TLSConnection)
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.TLSConnection')
+    def test_create_ssh_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test('ssh', SSHConnection)
+
+        connection = create_connection('ssh', port=123)
+        self.assertTrue(isinstance(connection, SSHConnection))
```

Comment:
Sorry I don't get why you are using the `perform_create_connection_test` method and afterwards very similar code. So either the method should be extended to be more generic or (my personal preference) just remove it completely and copy the code to all test methods. The method is only a two liner and therefore I think the abstraction is not necessary.

Reply:
Actually the lines 492 and 493 were created when I haven't created the `perform_create_connection_test` method. I have forgot to delete these lines after refactoring 492 and 493 into `perform_create_connection_test` :D I will delete them.. :)